models/purple_alien/src/offline_evaluation/evaluate_model.py

```diff
@@ -161,96 +87,98 @@ def get_posterior(model, views_vol, config, device):
+            pickle.dump(metric_dict, file)
 
-    # tell wandb to get started
-    with wandb.init(project=project, entity="nornir", config=config): # project and config ignored when runnig a sweep
+        with open(f'{PATH_GENERATED}/test_vol_{config.time_steps}_{config.run_type}_{config.model_time_stamp}.pkl', 'wb') as file: # make it numpy
+            pickle.dump(full_tensor.cpu().numpy(), file)
 
-        wandb.define_metric("monthly/out_sample_month")
-        wandb.define_metric("monthly/*", step_metric="monthly/out_sample_month")
+        print('Posterior dict, metric dict and test vol pickled and dumped!')
 
-        # access all HPs through wandb.config, so logging matches execution!
-        config = wandb.config
 
-        views_vol = get_data(config)
+    else:
+        print('Running sweep. NO posterior dict, metric dict, or test vol pickled+dumped')
 
-        # computerome artifacts path
-        #artifacts_path = f"/home/projects/ku_00017/people/simpol/scripts/conflictNet/artifacts"
-        
-        # fimbulthul artifacts path
-        artifacts_path = config.path_artifacts # f"/home/simmaa/HydraNet_001/artifacts" # should be in config <---------------------------------------------------------------------------------------------------
-
-        model = torch.load(f"{artifacts_path}/calibration_model.pt") # you rpolly need configs for both train and test...
-
-        get_posterior(model, views_vol, config, device) # actually since you give config now you do not need: time_steps, run_type, is_sweep,
-        print('Done testing')
-
-        return(model)
-
-
-if __name__ == "__main__":
-
-    wandb.login()
+    # could be a function in utils_wandb....
+    wandb.log({f"{config.time_steps}month_mean_squared_error": np.mean(mse_list)})
+    wandb.log({f"{config.time_steps}month_average_precision_score": np.mean(ap_list)})
+    wandb.log({f"{config.time_steps}month_roc_auc_score": np.mean(auc_list)})
+    wandb.log({f"{config.time_steps}month_brier_score_loss":np.mean(brier_list)})
 
-    time_steps_dict = {'a':12,
-                       'b':24,
-                       'c':36,
-                       'd':48,}
 
-    time_steps = time_steps_dict[input('a) 12 months\nb) 24 months\nc) 36 months\nd) 48 months\nNote: 48 is the current VIEWS standard.\n')]
+def handle_evaluation(config, device, views_vol, PATH_ARTIFACTS, artifact_name=None):
 
-    model_type_dict = {'a' : 'calibration', 'b' : 'testing'}
-    model_type = model_type_dict[input("a) Calibration\nb) Testing\n")]
-    print(f'Run type: {model_type}\n')
+    # if an artifact name is provided through the CLI, use it. Otherwise, get the latest model artifact based on the run type
+    if artifact_name:
+        print(f"Using (non-default) artifact: {artifact_name}")
+        
+        # If it lacks the file extension, add it
+        if not artifact_name.endswith('.pt'):
```

Comment:
Same idea about file extension

Reply:
Yeah, since this is purple alien specific in the model src I'll leave it for now. But also yes, we could abstract some of this out at some point